src/contracts/IBridgeV3.ts

```diff
@@ -1,26 +1,28 @@
-import { Contract, EventData } from 'web3-eth-contract';
+import { Contract, EventLog } from 'web3-eth-contract';
 import { IBridge } from './IBridge';
+import {ContractAbi} from "web3";
 
 interface MappedTokensParams {
   originalTokenAddress: string;
 }
 
 export class IBridgeV3 implements IBridge {
-  bridgeContract: Contract;
+  bridgeContract: Contract<ContractAbi>;
 
-  constructor(bridgeContract: Contract) {
+  constructor(bridgeContract: Contract<ContractAbi>) {
     this.bridgeContract = bridgeContract;
   }
 
-  getFederation(): Promise<string> {
-    return this.bridgeContract.methods.getFederation();
+  async getFederation(): Promise<string> {
+    return this.bridgeContract.methods.getFederation() as unknown as string;
```

Comment:
this is because you're using `ContractAbi` as type parameter for `Contract` maybe is worth to consider typechain? In this case you don't have problem since its a trivial case but this casting might lead to errors later, since using typechain (or providing the abi types in any way) may involve some time I'll let up to you to decide

Reply:
(same comment in general for similar cases)